CMake/deps/asan.cmake

```diff
@@ -0,0 +1,9 @@
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  message(STATUS "Enabling address sanitizer")
+  target_compile_options(clio_tests PRIVATE -g -fsanitize=address -fno-omit-frame-pointer) 
+  target_link_options (clio_tests PRIVATE -g -fsanitize=address -fno-omit-frame-pointer)
```

Comment:
Are these required to be link options also?

Reply:
Chances are they are not. I have not tried without but it could work just fine too. In some old project of mine i had them in the link options so i thought maybe i have done that for a reason before - can't really remember.